app/validators/program_validator.rb

```diff
@@ -1,17 +1,36 @@
 # frozen_string_literal: true
 
 class ProgramValidator < BaseValidator
-  def validate(record)
-    duplicates_exist = Program.where(['description = ? AND facility_code = ? AND id != ?',
-                                      record.description, record.facility_code, record.id]).any?
-    record.errors[:base] << non_unique_error_msg(record) if duplicates_exist
+  def self.after_import_batch_validations(validation_warnings)
+    duplicate_facility_description_results.each do |record|
+      record.errors[:base] << non_unique_error_msg(record)
+      add_warning_message(record, validation_warnings)
+    end
 
-    facility_code_in_weam?(record)
+    missing_facility_in_weam.each do |record|
+      record.errors[:base] << missing_facility_error_msg(record)
+      add_warning_message(record, validation_warnings)
+    end
   end
 
-  private
+  def self.add_warning_message(record, validation_warnings)
```

Comment:
Can we rename this method to something like `add_record_to_failed_instances` (and rename parameters as appropriate) since it's not actually adding a warning message?

Reply:
renamed